### ISSUE TITLE
Port names in `@relation` macro

### ISSUE THREAD

**Issue by epatters** [AlgebraicJulia/Catlab.jl#314]
Extend the `@relation` macro to allow port names. For example, the following relation would represent paths of length 2 in a graph:

```julia
@relation (start=u, stop=v) where (u::V, v::V, w::V) begin
  E(src=u, tgt=v)
  E(tgt=v, tgt=w)
end
```

This feature should make it easier to write conjunctive queries in the database setting, such as for the bicategory of spans over a C-set or for SQL using AlgebraicRelations.jl (cc @bosonbaas).

**Comment by bosonbaas** [comment]
This would be an excellent feature! Adding in the port names will definitely make the conversion to conjunctive queries significantly simpler to both implement and use.

**Comment by epatters** [comment]
Resolved by #344.
